sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -1011,11 +1011,29 @@ private Block BuildImplicitElementAccessExpression(ImplicitElementAccessSyntax e
         private Block BuildInvocationLikeExpression(ExpressionSyntax parent, Block currentBlock,
             ExpressionSyntax child, IEnumerable<ArgumentSyntax> arguments)
         {
-            var args = arguments == null
-                ? Enumerable.Empty<ExpressionSyntax>()
-                : arguments.Select(a => a.Expression);
+            currentBlock.ReversedInstructions.Add(parent);
+            var isNameof = parent is InvocationExpressionSyntax invocation
+                && invocation.IsNameof(this.semanticModel);
 
-            return BuildSimpleNestedExpression(parent, currentBlock, new[] { child }.Concat(args));
+            if (isNameof)
+                return currentBlock;
+
+            foreach (var arg in arguments.Reverse())
+            {
+                if (arg.RefOrOutKeyword.IsKind(SyntaxKind.RefKeyword))

```

Comment:
why are you putting `ref` arguments first and non-`ref` after? please add comments

Reply:
ill add a comment. ref argument is put last not first, since it is in reverse ;)